### PR TITLE
[release/6.0] Bump a few package versions

### PR DESCRIPTION
**Pull request by dougbu**
- small part of https://github.com/dotnet/dnceng/issues/345
- bump Azure.Data.Tables to v12.8.1
- replace Microsoft.IdentityModel.Clients.ActiveDirectory with Microsoft.Identity.Client
  - part of https://github.com/dotnet/dnceng/issues/629
- bump NuGet.* to v6.7.0
  - part of https://dev.azure.com/dnceng/internal/_workitems/edit/5128
  - part of https://dev.azure.com/dnceng/internal/_workitems/edit/5129
  - left both `$(NuGetVersioningVersion)` and `$(NuGetVersion)` to minimize disruption
    - see #12333 change in 'main' for a counter-example
  - react to deprecation of `ContentItemCollection.FindItemGroups(...)`
  - react to removal of `SignatureVerificationProviderFactory`
  - react to `VerifySignaturesResult` API changes (`Valid` to `IsValid`)
- bump NewtonSoft.Json to v13.0.3
- nits:
  - remove unused `$(...Version)` properties
  - remove a bit of dead code and unused `using`s in changed files

### To double check:

* [x] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -44,13 +39,12 @@
     <MoqVersion>4.8.3</MoqVersion>
     <MonoOptionsVersion>5.3.0.1</MonoOptionsVersion>
     <McMasterExtensionsCommandLineUtils>2.3.0</McMasterExtensionsCommandLineUtils>
-    <NewtonsoftJsonVersion>13.0.1</NewtonsoftJsonVersion>
+    <NewtonsoftJsonVersion>13.0.3</NewtonsoftJsonVersion>
     <SystemTextJsonVersion>4.7.0</SystemTextJsonVersion>
-    <NuGetVersioningVersion>4.4.0</NuGetVersioningVersion>
-    <NuGetVersion>5.6.0-preview.2.6489</NuGetVersion>
+    <NuGetVersioningVersion>6.7.0</NuGetVersioningVersion>
```
Comment (dougbu) [inline eng/Versions.props:44]: hoping CI build and arcade-validation will confirm lack of behaviour breaking changes in moves from 4.4.0 and 4.7.0 to here.

File: src/Microsoft.DotNet.Deployment.Tasks.Links/src/AkaMSLinksManager.cs [hunk 2]
```diff
@@ -379,16 +378,34 @@ public async Task DeleteLinksAsync(List<string> linksToDelete)
...
+            Documentation of new model (Microsoft Authentication Library for .NET, MSAL.NET) recommends caching
+            tokens to disk e.g.,
+            https://learn.microsoft.com/en-ca/azure/active-directory/develop/msal-net-migration-confidential-client?tabs=daemon
+            and
+            https://learn.microsoft.com/en-us/dotnet/api/microsoft.identity.client.cacheoptions.usesharedcache?view=msal-dotnet-latest#remarks
+
+            Specific requirements for relatively long-lived desktop apps (like this build task) are however unclear.
+            Disk cache appears to matter most when applications are restarted frequently i.e., it's about not losing
+            tokens when the app restarts.
+            ??? Reviewers: Thoughts on adding a disk cache at this time?
```
Comment (dougbu) [inline src/Microsoft.DotNet.Deployment.Tasks.Links/src/AkaMSLinksManager.cs:403]: calling this out. I'm also thinking of removing this comment block if we decide a disk cache isn't needed

File: src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs [hunk 3]
```diff
@@ -31,20 +28,25 @@ public override SignatureVerificationResult VerifySignature(string path, string
             return svr;
         }
 
+        // This method and SignatureVerificationResult.IsSigned are slightly misnamed. Signature validity is just as
+        // important as signature existence. The new VerifySignatureResult.IsSigned property would _not_ be correct
+        // to use here.
         private bool IsSigned(string path)
```
Comment (dougbu) [inline src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs:34]: from everything I saw in NuGet code, semantics of this method should be 💯% identical after these changes

**Comment by mmitche** [comment]
Looks like this fixed some CG issues, but introduced others: https://dev.azure.com/dnceng/internal/_build/results?buildId=2267280&view=logs&j=316d5c15-0c50-544e-8051-e6b14a1ab674&t=976ed21e-c736-59e1-f431-046bb8d7ebcd

**Comment by dougbu** [comment]
thanks for merging this and #14001. I'll look at the problems once we get helix-machines in a better state and I've addressed the issues in #14003

**Pull request merged**